### PR TITLE
Redesign build process

### PR DESCRIPTION
**Pull request by calvix**
So this is a huge PR but it did not make sense to separate it.
Changes:
1.  separating mayu into three repositories

https://github.com/giantswarm/mayu/
https://github.com/giantswarm/mayuctl
https://github.com/giantswarm/mayu-infopusher

Instead of using binaries of `mayuctl` and `mayu-infopusher` its advised to use theirs docker images, templates has been changed accordingly 

2.  Redesign of build and deploy process - no more `Makefile` and `bin-dist`, building is now via `go build` and for deploying you can easily use quay docker image, all custom configurations and  required files  can be mounted via docker from host.

3. cleaning - removing old and unused stuff, removed any templates that had giantswarm internal configuration, replaced with simple examples

4. moved from old outdated `ubuntu:willy` to `alpine` docker image

5. few changes to code to  be able to build via architect (changes for `go vet`, `go fmt`, added `version` and `--help` commands

ping @teemow @giantswarm/team-positive @JosephSalisbury @xh3b4sd

### CODE REVIEW

**Comment by xh3b4sd** [comment]
> it did not make sense to separate it

Just for the record. In theory, the list you wrote down in the PR description is the conceptual separation of different changes and could have been addressed by separate PRs.

File: Makefile [hunk 1]
```diff
@@ -1,79 +0,0 @@
...
-release386: bin-dist
-		rm -rf bin
-		@GOOS=linux; GOARCH=386 $(MAKE) bin/mayu
-		@GOOS=linux; GOARCH=386 $(MAKE) bin/mayuctl
-		cp bin/mayu* bin-dist
-		cd bin-dist && rm -f $(PROJECT).*-linux-i386.tar.gz && tar czf $(PROJECT).$(VERSION)-linux-i386.tar.gz --exclude='*.tar.gz' *
-
-clean:
-		rm -rf .gobuild bin helpers/infopusher
-		cd infopusher ; make clean
```
Comment (xh3b4sd) [inline Makefile]: I love that this is gone.

File: circle.yml [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
-        - cd $GS_WD && go build -a -v -tags netgo -ldflags "-X main.projectVersion=$CIRCLE_SHA1 -X main.projectBuild=$CIRCLE_SHA1" -o bin/mayu github.com/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME
-        - cd $GS_WD && go build -a -v -tags netgo -ldflags "-X main.projectVersion=$CIRCLE_SHA1 -X main.projectBuild=$CIRCLE_SHA1" -o bin/mayuctl github.com/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/mayuctl
+  override:
+    - ./architect build
+
+deployment:
+  master:
+    branch: master
+    commands:
+     - ./architect deploy
```
Comment (xh3b4sd) [inline circle.yml]: I like that.

File: mayu.service [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
 
 [Service]
-Restart=always
+Restart=on-failure
+RestartSec=1
+StartLimitInterval=300s
+StartLimitBurst=3
 TimeoutStartSec=0
-ExecStart=/usr/bin/docker run --rm --cap-add=NET_ADMIN --net=host \
+Environment="IMAGE=quay.io/giantswarm/mayu:c832dd12fa8f5cc77101ae05820106ace18504f7"
```
Comment (xh3b4sd) [inline mayu.service]: For my interest, why is this unit file in here? I feel like this should just be an example file with some placeholders. I can imagine people will just run it and wonder why there is a specific version being used and not something like the latest.

File: pxemgr/pxemanager.go [hunk 4]
```diff
@@ -386,8 +383,7 @@ func (mgr *pxeManagerT) getNextInternalIP() net.IP {
 		currentIP = incIP(currentIP)
 	}
 
-	panic(errors.New("unable to get a free ip"))
-	return net.IP{}
+	//panic(errors.New("unable to get a free ip"))
```
Comment (xh3b4sd) [inline pxemgr/pxemanager.go]: What happens here? Can we remove it? Should we use it?

File: templates/last_stage_cloudconfig.yaml [hunk 5]
```diff
@@ -9,109 +9,145 @@ coreos:
...
       RemainAfterExit=yes
 
       [Install]
       WantedBy=multi-user.target
-  {{end}}
   - name: etcd2.service
-    enable: true
-    command: start
+    enable: false
+    command: stop
```
Comment (xh3b4sd) [inline templates/last_stage_cloudconfig.yaml]: Should we mask it to have it permanently disabled?

File: pxemgr/pxemanager.go (hunk 4 above)
Comment (calvix) [inline pxemgr/pxemanager.go]: I cant really say what is happening there, it was `unreachable code` from go vet, we can remove it

File: README.md [hunk 6]
```diff
@@ -37,12 +36,10 @@ Developing Mayu requires the following tools to be installed.
 
 ## Getting Mayu
 
-Download the latest release from here: https://github.com/giantswarm/mayu/releases/latest
+Download the latest release from here: get latest docker image here: https://quay.io/repository/giantswarm/mayu
```
Comment (JosephSalisbury) [inline README.md]: I'd just go with 

`Get the latest Docker image here: https://quay.io/repository/giantswarm/mayu`

File: README.md [hunk 7]
```diff
@@ -58,6 +55,10 @@ Check code documentation: [godoc](https://godoc.org/github.com/giantswarm/mayu)
 
 - Future directions/vision
 
+## Changes
```
Comment (JosephSalisbury) [inline README.md]: I'd remove this, or write it in a general way.

e.g:
```
Use the latest image from quay.io, and mount the necessary files to the container
```

File: VERSION [hunk 8]
```diff
@@ -1 +1 @@
-0.12.6+git
\ No newline at end of file
+0.13.0+git
```
Comment (JosephSalisbury) [inline VERSION]: I think we can remove this file now?

File: docs/compiling.md [hunk 9]
```diff
@@ -1,17 +1,15 @@
...
+In order to compile Mayu you need to have `golang`  installed.
 
-To compile all binaries of the projects running a simple `make` is sufficient.
+To compile all binaries of the projects running a simple `go build` is sufficient.
 
 ## Updating vendors
 
-To update the vendored libraries used by Mayu's binaries you need to have
-`make` and [glide](https://github.com/Masterminds/glide) installed.
+To update the vendored libraries used by Mayu's binaries you need to havve [glide](https://github.com/Masterminds/glide) installed.
```
Comment (JosephSalisbury) [inline docs/compiling.md]: havve -> have

File: hostmgr/host.go [hunk 10]
```diff
@@ -28,7 +28,7 @@ type Host struct {
 	IPMIAddr           net.IP    `json:",omitempty"`
 	Hostname           string    `json:",omitempty"`
 	MachineID          string    `json:",omitempty"`
-	ConnectedNIC       string    `json:",omitempty`
+	ConnectedNIC       string    `json:",omitempty"`
```
Comment (JosephSalisbury) [inline hostmgr/host.go]: It worries me how long this has been wrong :D

File: mayu.service (hunk 3 above)
Comment (JosephSalisbury) [inline mayu.service]: Yeah, it might be an idea to change this to `mayu.service.dist`

File: templates/last_stage_cloudconfig.yaml (hunk 5 above)
Comment (calvix) [inline templates/last_stage_cloudconfig.yaml]: masked

**Comment by calvix** [comment]
updated according to requested changes :)

**Comment by xh3b4sd** [comment]
+1 on strategy brainstorming. When it looks like something is too big or weird we can always talk and see what makes sense. Smaller PRs are more easy to review and the review quality is way better when there are less changes that do not have anything in common conceptually wise.

File: pxemgr/pxemanager.go (hunk 4 above)
Comment (kopiczko) [inline pxemgr/pxemanager.go]: I'm late here, but: for loop above `for ; ; ipLessThanOrEqual(currentIP, rangeEnd) {` is infinite. I think intention there was `for ipLessThanOrEqual(currentIP, rangeEnd)`. Then `panic` would be reachable.

**Pull request merged**